### PR TITLE
working project

### PR DESCRIPTION
**Pull request by Naher123**
project complete
but did not yet get sparkmax simulation support working.

### CODE REVIEW

File: camp/nfutsum/Twisty/src/main/java/frc/robot/RobotContainer.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+ * "declarative" paradigm, very little robot logic should actually be handled in the {@link Robot}
+ * periodic methods (other than the scheduler calls). Instead, the structure of the robot (including
+ * subsystems, commands, and button mappings) should be declared here.
+ */
+public class RobotContainer {
+  // The robot's subsystems and commands are defined here...
+  private final NaherNeoMotor myMotor = new NaherNeoMotor();
+
+  private final StopTheMotor m_autoCommand = new StopTheMotor(myMotor);
+  private final Joystick myJoystick = new Joystick(0);
```
Comment (josieammer) [inline camp/nfutsum/Twisty/src/main/java/frc/robot/RobotContainer.java:29]: The joystick ID should be stored in Constants.java

File: camp/nfutsum/Twisty/src/main/java/frc/robot/RobotContainer.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+   * Use this to pass the autonomous command to the main {@link Robot} class.
+   *
+   * @return the command to run in autonomous
+   */
+  public Command getAutonomousCommand() {
+    // An ExampleCommand will run in autonomous
+    return m_autoCommand;
+  }
+
+  public void simulationInit(){
```
Comment (josieammer) [inline camp/nfutsum/Twisty/src/main/java/frc/robot/RobotContainer.java:57]: missing definition comment bloc - applies to any public method, constructor and class

File: camp/nfutsum/Twisty/src/main/java/frc/robot/commands/MotorGo.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {}
+
+  // Called every time the scheduler runs while the command is scheduled.
+  @Override
+  public void execute() {
+    double s = m_joystick.getTwist();
+    //.out.println("speed: " + s);
```
Comment (josieammer) [inline camp/nfutsum/Twisty/src/main/java/frc/robot/commands/MotorGo.java:38]: remove debug print code.

File: camp/nfutsum/Twisty/src/main/java/frc/robot/commands/StopTheMotor.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.commands;
+
+import frc.robot.subsystems.NaherNeoMotor;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+
+/** An example command that uses an example subsystem. */
```
Comment (josieammer) [inline camp/nfutsum/Twisty/src/main/java/frc/robot/commands/StopTheMotor.java:10]: update the comment block. and all the ones below.

File: camp/nfutsum/Twisty/src/main/java/frc/robot/subsystems/NaherNeoMotor.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+package frc.robot.subsystems;
+
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+
+public class NaherNeoMotor extends SubsystemBase {
```
Comment (josieammer) [inline camp/nfutsum/Twisty/src/main/java/frc/robot/subsystems/NaherNeoMotor.java:13]: missing definition comment bloc - applies to any public method, constructor and class

**Review by josieammer — CHANGES_REQUESTED** [review]

File: camp/nfutsum/Twisty/src/main/java/frc/robot/RobotContainer.java (hunk 1 above)
Comment (mmcadamsms) [reply camp/nfutsum/Twisty/src/main/java/frc/robot/RobotContainer.java:29]: +1

File: camp/nfutsum/Twisty/src/main/java/frc/robot/commands/MotorGo.java [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+package frc.robot.commands;
+
+import frc.robot.subsystems.NaherNeoMotor;
+import edu.wpi.first.wpilibj.Joystick;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+
+/** An example command that uses an example subsystem. */
+public class MotorGo extends CommandBase {
+  @SuppressWarnings({"PMD.UnusedPrivateField", "PMD.SingularField"})
```
Comment (mmcadamsms) [inline camp/nfutsum/Twisty/src/main/java/frc/robot/commands/MotorGo.java:14]: please provide a comment on why to suppress or remove suppression

File: camp/nfutsum/Twisty/src/main/java/frc/robot/commands/MotorGo.java [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {}
+
+  // Called every time the scheduler runs while the command is scheduled.
+  @Override
+  public void execute() {
+    double s = m_joystick.getTwist();
+    //.out.println("speed: " + s);
+    // this.m_subsystem.setSpeed(m_joystick.getTwist());
```
Comment (mmcadamsms) [inline camp/nfutsum/Twisty/src/main/java/frc/robot/commands/MotorGo.java:39]: removed line not being used

File: camp/nfutsum/Twisty/src/main/java/frc/robot/RobotContainer.java [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+    REVPhysicsSim.getInstance().addSparkMax(myMotor.getInstance(), DCMotor.getNEO(1));
+  }
+
+  public void simulationPeriodic(){
```
Comment (mmcadamsms) [inline camp/nfutsum/Twisty/src/main/java/frc/robot/RobotContainer.java:61]: repeat josie's comment from above for this method - add a comment block

File: camp/nfutsum/Twisty/src/main/java/frc/robot/commands/MotorGo.java (hunk 6 above)
Comment (Naher123) [reply camp/nfutsum/Twisty/src/main/java/frc/robot/commands/MotorGo.java:14]: I don't think I needed that line so I removed it

**Pull request closed without merging**